.github/workflows/ci.yml

```
@@ -140,15 +140,15 @@ jobs:
         run: ./run-tests.sh --check-sphinx
 
   python-tests:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
       - name: Setup Python
         uses: actions/setup-python@v5
         with:
-          python-version: 3.8
+          python-version: 3.12
```

nitpicking: maybe let's quote the version in all the invocations of `setup-python`, as it's done in all the examples at https://github.com/actions/setup-python ?
```suggestion
          python-version: "3.12"
```